lib/Sema/CSBindings.cpp

```diff
@@ -1567,18 +1567,27 @@ void PotentialBindings::retract(Constraint *constraint) {
   EquivalentTo.remove_if(hasMatchingSource);
 }
 
-LiteralBindingKind BindingSet::getLiteralKind() const {
-  LiteralBindingKind kind = LiteralBindingKind::None;
-
+SmallVector<KnownProtocolKind> BindingSet::getProtocolKind() const {
+  SmallVector<KnownProtocolKind> kind;
   for (const auto &literal : Literals) {
     auto *protocol = literal.first;
     const auto &info = literal.second;
 
     // Only uncovered defaultable literal protocols participate.
     if (!info.viableAsBinding())
       continue;
+    
+    if (protocol->getKnownProtocolKind().hasValue())
+      kind.push_back(protocol->getKnownProtocolKind().getValue());
```

Comment:
nit: Could be like this?  
```suggestion
    if (auto kind = protocol->getKnownProtocolKind()) {
      kind.push_back(kind);
    }
```

Reply:
Similarly, `getKnownProtocolKind()` returns an Optional<>, so I check `hasValue()` before `getValue()`.

Reply:
@amritpan, @LucianoPAlmeida suggests to use a common pattern - using `if` here would check `hasValue` implicitly and if it returns `true` execute the block. This works because llvm::Optional declares `operator bool` overload - https://llvm.org/doxygen/classllvm_1_1Optional.html#ae6fda74944be767b86f133707f0496da which is then called by `if` under the  hood - https://en.cppreference.com/w/cpp/language/if

Reply:
Yeah, I also missed the `*` on suggestion. Should be 
```cpp
if (auto kind = protocol->getKnownProtocolKind()) {
      kind.push_back(*kind);
}
```
But I see, you used `getValue()` already, and between the two is more personal preference :)